src/words/TokTok.jl

```diff
@@ -0,0 +1,255 @@
+const CLOSE_PUNCT = (
+        "\u0f3b", "\u0f3d", "\u169c", "\u2046", "\u207e", "\u208e", "\u232a", "\uff63",
+        "\u2769", "\u276b", "\u276d", "\u276f", "\u2771", "\u2773", "\u2775", "\u27c6",
+        "\u27e7", "\u27e9", "\u27eb", "\u27ed", "\u27ef", "\u2984", "\u2986", "\u2988",
+        "\u298a", "\u298c", "\u298e", "\u2990", "\u2992", "\u2994", "\u2996", "\u2998",
+        "\u29d9", "\u29db", "\u29fd", "\u2e23", "\u2e25", "\u2e27", "\u2e29", "\u3009",
+        "\u300b", "\u300d", "\u300f", "\u3011", "\u3015", "\u3017", "\u3019", "\u301b",
+        "\u301e", "\u301f", "\ufd3f", "\ufe18", "\ufe36", "\ufe38", "\ufe3a", "\ufe3c",
+        "\ufe3e", "\ufe40", "\ufe42", "\ufe44", "\ufe48", "\ufe5a", "\ufe5c", "\ufe5e",
+        "\uff09", "\uff3d", "\uff5d", "\uff60")
+
+# This is the \p{Close_Punctuation} from Perl's perluniprops
+const CURRENCY_SYM = (
+        "\u00a2", "\u00a3", "\u00a4", "\u00a5", "\u058f", "\u060b", "\u09f2", "\u09f3",
+        "\u0af1", "\u0bf9", "\u0e3f", "\u17db", "\u20a0", "\u20a1", "\u20a2", "\u20a3",
+        "\u20a4", "\u20a5", "\u20a6", "\u20a7", "\u20a8", "\u20a9", "\u20aa", "\u20ab",
+        "\u20ac", "\u20ad", "\u20ae", "\u20af", "\u20b0", "\u20b1", "\u20b2", "\u20b3",
+        "\u20b4", "\u20b5", "\u20b6", "\u20b7", "\u20b8", "\u20b9", "\u20ba", "\ua838",
+        "\ufdfc", "\ufe69", "\uff04", "\uffe0", "\uffe1", "\uffe5", "\uffe6", "\u09fb")
+
+const rules_atoms = Tuple(Iterators.flatten([
+        CURRENCY_SYM,
+        FUNKY_PUNCT_1,
+        FUNKY_PUNCT_2,
+        EN_EM_DASHES,
+        PROB_SINGLE_QUOTES,
+        OPEN_PUNCT,
+        CLOSE_PUNCT]))
+
+const rules_replaces = Tuple(Iterators.flatten([
+        [AMPERCENT],
+        [TAB],
+        [PIPE],
+        [STUPID_QUOTES_1],
+        [STUPID_QUOTES_2]]))
+
+
+"""
+    totok_tokenize(instring::AstractString)
+This tokenizer is a simple, general tokenizer, where the input has one sentence per line; thus only final period is tokenized.
+Tok-tok has been tested on and gives reasonably good results for English, Persian, Russian, Czech, French, German, Vietnamese,
+Tajik, and a few others.
+"""
+function toktok_tokenize(instring::AbstractString)
+    ts = TokenBuffer(instring)
+    isempty(ts.input) && return ts.tokens
+
+    effective_end = handle_final_periods(ts)
+
+    while !isdone(ts) && ts.idx <= effective_end
+	if string(ts.input[ts.idx]) == NON_BREAKING[1]
+		ts.input[ts.idx] = NON_BREAKING[2]
+	end
+
+        url_handler4(ts) ||   # these url handlers have priority over others
+        url_handler3(ts) ||
+        url_handler2(ts) ||
+        url_handler1(ts) ||
+	repeated_character_seq(ts, ',', 2) ||
+        repeated_character_seq(ts, '-', 2) ||
+        repeated_character_seq(ts, '.', 2) ||        
+	number(ts) ||
+        spaces(ts) ||      # Handles ONE_SPACE rules from original toktok perl script
+	replaces(ts, rules_replaces) ||    # most expensive steps, keep low priority
```

Comment:
Found that these were the most expensive steps by profiling, thus reduced there priority to increase performance of code. Change in it's priority won't effect the tokenizer.